security/pkg/pki/ra/common.go

```diff
@@ -86,9 +89,76 @@ func ValidateCSR(csrPEM []byte, subjectIDs []string) bool {
 			return false
 		}
 	}
+
+	// Verify Istio CSR was created by Istio-Proxy by generating template using same the
+	// same process as performed on the client and comparing the generated CSR with the
+	// received CSR.
+
+	// Convert csrIDs to Host comma separated string
+	hosts := strings.Join(csrIDs, ",")
+	// genCSRTemplate is the expected CSR template. The template's extensions are marshaled
+	// in the ExtraExtensions field. CreateCertificateRequest would normally add the SAN extensions
+	// from ExtraExtensions to the Extensions field. However, we are only generating the
+	// template here and not the actual CSR since we do not know the exact signing mechanisms
+	// of the client. Additionally, if we compared extensions the the generated CSR and the original
+	// CSR would match since the hosts where constructed from the extracted SAN extensions.
+	genCSRTemplate, err := util.GenCSRTemplate(util.CertOptions{Host: hosts})
```

Comment:
If the hangup is really just Subject.Org, we can do

`util.GenCSRTemplate(util.CertOptions{Host: hosts, Org: csr.Subject.Organization})`

and compare.